chaosinstana/probes.py

```diff
@@ -0,0 +1,51 @@
+from chaoslib.exceptions import ActivityFailed
+from chaoslib.types import Configuration, Secrets
+
+from logzero import logger
+
+import dateparser
+import datetime
+
+from .api import get_all_events
+
+__all__ = ["get_all_events_in_window", "convert_time_string_to_millis_string"]
+
+
+def convert_time_string_to_millis_string(source_time_string: str) -> str:
+    """
+    Parse the time string to check its a valid time, first see if the
+    string is an int and treats as a milliseconds since epoch, if
+    the string is not an int uses dataparser.
+    """
+    try:
+        source_time_int = int(source_time_string)
+        source_dt = datetime.datetime.fromtimestamp(source_time_int / 1000)
+        timestamp_int = int(source_dt.timestamp() * 1000)
+        return str(timestamp_int)
+    except ValueError:
+        # source_time_string not an int
+        start_dt = dateparser.parse(source_time_string)
+        if (start_dt):
+            return str(int(start_dt.timestamp()) * 1000)
+        else:
+            logger.error("Invalid date string provided")
+            raise ActivityFailed("Invalid date string provided")
+
+
+def get_all_events_in_window(from_time: str = None, to_time: str = None,
+                             configuration: Configuration = None,
+                             secrets: Secrets = None) -> str:
+    """
+    get all events from instana within a time window, given by the from_time
```

Comment:
Consider this is part of our documentation, so make sure this is geared towards the people writing the experiments.

https://docs.chaostoolkit.org/drivers/overview/